src/Tests/dotnet-workload-install.Tests/GivenNetSdkManagedWorkloadInstall.cs

```diff
@@ -300,7 +300,69 @@ public void GivenManagedInstallItCanInstallManifestVersion()
+            var (dotnetRoot, installer, nugetInstaller) = GetTestInstaller();
+            var version = "6.0.100";
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"), "Xamarin.Android.Sdk");
+            var cachePath = Path.Combine(dotnetRoot, "MockCache");
+            installer.DownloadToOfflineCache(packInfo, cachePath);
+
+            var mockNugetInstaller = nugetInstaller as MockNuGetPackageDownloader;
+            mockNugetInstaller.DownloadCallParams.Count.Should().Be(1);
+            mockNugetInstaller.DownloadCallParams[0].ShouldBeEquivalentTo((new PackageId(packInfo.Id), new NuGetVersion(packInfo.Version), cachePath));
+            Directory.Exists(cachePath).Should().BeTrue();
+
+            // Should have only been downloaded, not installed
+            mockNugetInstaller.ExtractCallParams.Count.Should().Be(0);
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", "v1", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeFalse();
+            Directory.Exists(packInfo.Path).Should().BeFalse();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanInstallPacksFromOfflineCache()
+        {
+            var (dotnetRoot, installer, nugetInstaller) = GetTestInstaller();
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"), "Xamarin.Android.Sdk");
+            var version = "6.0.100";
+            var cachePath = Path.Combine(dotnetRoot, "MockCache");
+
+            // Write mock cache
+            Directory.CreateDirectory(cachePath);
+            var nupkgPath = Path.Combine(cachePath, $"{packInfo.ResolvedPackageId}.{packInfo.Version}.nupkg");
+            File.Create(nupkgPath);
+
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version), cachePath);
+            var mockNugetInstaller = nugetInstaller as MockNuGetPackageDownloader;
+            
+            // We shouldn't download anything, use the cache
+            mockNugetInstaller.DownloadCallParams.Count.Should().Be(0);
+
+            // Otherwise install should be normal
+            mockNugetInstaller.ExtractCallParams.Count.Should().Be(1);
+            mockNugetInstaller.ExtractCallParams[0].ShouldBeEquivalentTo((nupkgPath, Path.Combine(dotnetRoot, "metadata", "temp", $"{packInfo.Id}-{packInfo.Version}-extracted")));
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", "v1", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeTrue();
+            Directory.Exists(packInfo.Path).Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanErrorsWhenMissingOfflineCache()
```

Comment:
Could you also add an test for print URLs?

Reply:
I wrote one, it's in the other test file: https://github.com/dotnet/sdk/pull/17324/files#diff-c8bc89094b206b506f35f0cb9ac6e55591ae2468130efe0a19dab1810c0f0c1cR161